.taskcluster.yml

```diff
@@ -31,120 +31,137 @@ tasks:
         then: {$eval: 'event.ref[10:]' }
         else: "latest"
       else: "latest"
+
+    pullRequestAction:
+        $if: 'tasks_for[:19] == "github-pull-request"'
+        then: ${event.action}
+        else: false
+
+    isPullRequest:
+        $eval: 'tasks_for[:19] == "github-pull-request"'
+
   in:
-    - taskId: {$eval: as_slugid("code_checks")}
-      provisionerId: proj-relman
-      workerType: generic-worker-ubuntu-22-04
-      created: {$fromNow: ''}
-      deadline: {$fromNow: '1 hour'}
-      payload:
-        maxRunTime: 3600
-        image: python:3.10
-        env:
-          GIT_REPOSITORY: ${repository}
-          GIT_REVISION: ${head_rev}
-        command:
-          - sh
-          - -lxce
-          - "git clone --quiet ${repository} /src && cd /src && git checkout ${head_rev} -b taskboot &&
-            pip install --no-cache-dir --quiet -r requirements-tests.txt . &&
-            pre-commit run -a --show-diff-on-failure && pytest -v"
-      metadata:
-        name: TaskBoot code checks
-        description: Taskcluster boot utilities - code linting & unit tests
-        owner: bastien@mozilla.com
-        source: https://github.com/mozilla/task-boot
+    - $if: '(isPullRequest && pullRequestAction in ["opened", "reopened", "synchronize"]) || tasks_for == "github-push"'
```

Comment:
Could we have a single if that allows defining all tasks instead of one if for each task?

Reply:
good point, I'll check tomorrow
my json-e-fu might not be good enough :)

Reply:
updated, should look better now